File: articles/future-behavioral-economics-governance-policy/go/compliance_model.go
package main

import (
	"fmt"
	"math"
)

func logistic(x float64) float64 {
	return 1.0 / (1.0 + math.Exp(-x))
}

func complianceProbability(
	trust float64,
	salience float64,
	normSensitivity float64,
	burdenSensitivity float64,
	presentBias float64,
	adminBurden float64,
	reminderSalience float64,
	trustSignal float64,
	penaltyStrength float64,
) float64 {
	utility := 0.8*reminderSalience*salience +
		0.7*normSensitivity +
		1.0*trustSignal*trust +
		0.9*penaltyStrength -
		1.2*adminBurden*burdenSensitivity -
		0.7*presentBias*adminBurden

	return logistic(utility - 0.5)
}

func main() {
	p := complianceProbability(0.60, 0.55, 0.45, 0.60, 0.35, 0.12, 0.80, 0.80, 0.30)
	fmt.Printf("Synthetic compliance probability: %.3f\n", p)
}

File: articles/future-behavioral-economics-governance-policy/fortran/compliance_model.f90
program compliance_model
  implicit none

  real :: trust, salience, norm_sensitivity, burden_sensitivity
  real :: present_bias, admin_burden, reminder_salience
  real :: trust_signal, penalty_strength, utility, probability

  trust = 0.60
  salience = 0.55
  norm_sensitivity = 0.45
  burden_sensitivity = 0.60
  present_bias = 0.35
  admin_burden = 0.12
  reminder_salience = 0.80
  trust_signal = 0.80
  penalty_strength = 0.30

  utility = 0.8 * reminder_salience * salience &
          + 0.7 * norm_sensitivity &
          + 1.0 * trust_signal * trust &
          + 0.9 * penalty_strength &
          - 1.2 * admin_burden * burden_sensitivity &
          - 0.7 * present_bias * admin_burden

  probability = 1.0 / (1.0 + exp(-(utility - 0.5)))

  print *, "Synthetic compliance probability:", probability
end program compliance_model
